pkg/server/plugin/datastore/datastore.go

```diff
@@ -136,17 +132,17 @@ type ListBundlesResponse struct {
 type ListNodeSelectorsRequest struct {
 	// When enabled, read-only connection will be used to connect to database read instances. Some staleness of data will be observed.
 	TolerateStale bool
-	ValidAt       *timestamppb.Timestamp
+	ValidAt       *time.Time
```

Comment:
Same thing here (`time.Time` and use `IsZero()` to determine if it is set before applying)